kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.BulkDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+
+/**
+ * The Kube {@code Service} for a single virtual cluster.
+ * This is named like {@code ${cluster.name}-cluster-${KafkaProxy.metadata.name}}.
+ */
+@KubernetesDependent
+public class ClusterService
+        extends CRUDKubernetesDependentResource<Service, KafkaProxy>
+        implements BulkDependentResource<Service, KafkaProxy> {
+
+    public ClusterService() {
+        super(Service.class);
+    }
+
+    /**
+     * @return The {@code metadata.name} of the desired {@code Service}.
+     */
+    static String serviceName(KafkaProxy primary, Clusters cluster) {
+        Objects.requireNonNull(primary);
+        Objects.requireNonNull(cluster);
+        String serviceName = cluster.getName() + "-cluster-" + primary.getMetadata().getName();
```

Comment:
Arguably this is the wrong away around and it ought to be `${proxy}-cluster-${cluster}`. Currently we need the `-cluster-` in there to avoid a collision with the metrics service, which is `${proxy}-metrics`.